mpas_analysis/analysis_task_template.py

```diff
@@ -0,0 +1,217 @@
+'''
+This is an example analysis task to be used as a template for new tasks.
+It should be copied into one of the component folders (`ocean`, `sea_ice`,
+`land_ice`, etc.) and modified as needed.
+
+Specific parts or variable names that need to be modified by the developer are
+explicitly indicated. All the others are to be left unchanged.  Anything in
+comments or docstrings in triangle brackets <> indicates that it needs to be
+replaced with the appropriate variable name, text, etc. for the task.
+
+Instructions for creating a new analysis task:
+1. create a new task by copying this file to the appropriate folder (ocean,
+   sea_ice, etc.) and modifying it as described below.  Take a look at
+   mpas_analysis/shared/analysis_task.py for additional guidance.
+2. modify config.default (and possibly any machine-specific config files in
+   configs/<machine>)
+3. import new analysis task in mpas_analysis/<component>/__init__.py
+4. note, no changes need to be made to mpas_analysis/shared/analysis_task.py
+5. if necessary, update variable maps in
+   mpas_analysis/shared/variable_namelist_stream_maps/<component>_maps.py
+6. add new analysis task to run_analysis.py under build_analysis_list:
+      analyses.append(<component>.MyTask(config, myArg='argValue'))
+   will add a new object of the MyTask class to a list of analysis tasks
+   created in build_analysis_list.  Later on in run_analysis, it will first
+   go through the list to make sure each task needs to be generated
+   (by calling check_generate, which is defined in AnalysisTask), then, will
+   call setup_and_check on each task (to make sure the appropriate AM is on
```

Comment:
This is substantially more work than the other changes you requested.  I think it's conceptually useful to separate checking whether the setup is correct from running the task.  If we are going to request that each task do this, I don't see a big advantage in this being a helper function that each task runs at the beginning of `run` rather than being a separate function that they are required to implement.